internal/controller/driver_controller.go

```diff
@@ -756,6 +756,41 @@ func (r *driverReconcile) reconcileControllerPluginDeployment() error {
 								),
 							})
 						}
+						// ODF Snapshotter Sidecar Container
+						if r.images["odf-snapshotter"] != "" && r.driverType == CephFsDriverType {
```

Comment:
Having or not having an image in the image map is a poor indicator for desired state. We should have some sort of an ENV var controlling the need to run the additional side car

Reply:
That's what my initial approach was, but I updated it based on your previous comment [here](https://github.com/ceph/ceph-csi-operator/pull/246#discussion_r2113604352)

Reply:
You are right! resolving